libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -684,6 +688,321 @@ impl Greenlight {
         _ = self.inprogress_payments.fetch_sub(1, Ordering::Relaxed);
         res
     }
+
+    // pulls transactions from greenlight based on last sync timestamp.
+    // greenlight gives us the payments via API and for received payments we are looking for settled invoices.
+    async fn pull_transactions(
+        &self,
+        sync_state: &SyncState,
+        htlc_list: Vec<Htlc>,
+    ) -> NodeResult<(SyncState, Vec<Payment>)> {
+        let mut node_client = self.get_node_client().await?;
+        let mut new_sync_state = sync_state.clone();
+
+        // list invoices
+        let created_invoices = node_client
+            .list_invoices(cln::ListinvoicesRequest {
+                index: Some(ListinvoicesIndex::Created.into()),
+                start: Some(sync_state.list_invoices_index.created),
+                ..Default::default()
+            })
+            .await?
+            .into_inner();
+        if let Some(last) = created_invoices.invoices.last() {
+            new_sync_state.list_invoices_index.created = last.created_index()
+        }
+        let updated_invoices = node_client
```

Comment:
Currently we are adding extra call for invoices and payments (created, updated). Did you check perhaps querying by updated is enough? I mean what is the value of updated for a newly create invoice or payment? IF it is the same as the created timestamp than we should be good.

Reply:
Newly created pays/invoices are not returned when querying the updated index. It's really only for updates to previously created pays/invoices.